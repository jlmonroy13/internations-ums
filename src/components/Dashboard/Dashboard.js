import React from 'react';
import PropTypes from 'prop-types';
import Header from '../Header/';


function Dashboard({ children }) {
  return (
    <div className="dashboard">
      <div className="dashboard__container">
        <Header />
        {children}
      </div>
    </div>
  );
}

Dashboard.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Dashboard;
