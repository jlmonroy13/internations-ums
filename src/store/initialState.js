export default {
  authorization: {
    uid: '',
    isLoading: true,
  },
  events: {
    editMode: false,
    isMenuOpen: false,
    isNavOpen: false,
  },
};
