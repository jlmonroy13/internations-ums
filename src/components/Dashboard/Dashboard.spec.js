import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import initialState from '../../store/initialState';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { getFirebase } from 'react-redux-firebase';
import Dashboard from './Dashboard';

describe('<Dashboard />', () => {
  it('should render the dashboard component', () => {
    const middlewares = [thunk.withExtraArgument(getFirebase)];
    const mockStore = configureStore(middlewares);
    const store = mockStore(initialState);
    const wrapper = renderer.create(
      <Provider store={store}>
        <Dashboard>
          <span />
        </Dashboard>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
