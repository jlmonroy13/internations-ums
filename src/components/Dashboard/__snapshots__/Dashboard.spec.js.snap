// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render the dashboard component 1`] = `
<div
  className="dashboard"
>
  <div
    className="dashboard__container"
  >
    <header
      className="header"
    >
      <h1
        className="header__title"
      >
        <span
          className="icon icon--logo"
        />
         InterNations
         
        <span
          className="header__title--pink"
        >
          UMS
        </span>
      </h1>
      <span
        className="icon icon--menu header__btn"
        onClick={[Function]}
      />
    </header>
    <span />
  </div>
</div>
`;
